model/modelprocessor/errormessage_test.go

```diff
@@ -27,39 +27,90 @@ import (
 	"github.com/elastic/apm-data/model/modelprocessor"
 )
 
+// Copy the error message to the event message if event message is empty.
 func TestSetErrorMessage(t *testing.T) {
 	tests := []struct {
-		input   *modelpb.Error
-		message string
+		input    *modelpb.Error
+		expected string
 	}{{
-		input:   &modelpb.Error{},
-		message: "",
+		input:    &modelpb.Error{},
+		expected: "",
 	}, {
-		input:   &modelpb.Error{Log: &modelpb.ErrorLog{Message: "log_message"}},
-		message: "log_message",
+		input:    &modelpb.Error{Log: &modelpb.ErrorLog{Message: "log_message"}},
+		expected: "log_message",
 	}, {
-		input:   &modelpb.Error{Exception: &modelpb.Exception{Message: "exception_message"}},
-		message: "exception_message",
+		input:    &modelpb.Error{Exception: &modelpb.Exception{Message: "exception_message"}},
+		expected: "exception_message",
 	}, {
 		input: &modelpb.Error{
 			Log:       &modelpb.ErrorLog{},
 			Exception: &modelpb.Exception{Message: "exception_message"},
 		},
-		message: "exception_message",
+		expected: "exception_message",
 	}, {
 		input: &modelpb.Error{
 			Log:       &modelpb.ErrorLog{Message: "log_message"},
 			Exception: &modelpb.Exception{Message: "exception_message"},
 		},
-		message: "log_message",
+		expected: "log_message",
 	}}
-
 	for _, test := range tests {
 		batch := modelpb.Batch{{Error: test.input}}
 		processor := modelprocessor.SetErrorMessage{}
 		err := processor.ProcessBatch(context.Background(), &batch)
 		assert.NoError(t, err)
-		assert.Equal(t, test.message, batch[0].Message)
+		assert.Equal(t, test.expected, batch[0].Message)
+		assert.Equal(t, test.expected, batch[0].Error.Message)
 	}
+}
+
+// If event message is non-empty, do not override with error message.
+func TestSetEventMessage(t *testing.T) {
```

Comment:
Depending on the outcome of the above discussion, the tests can be consolidated by having expectedMessage and expectedErrorMessage fields.